src/custom/components/Tokens/TokensTableRow.tsx

```diff
@@ -103,6 +105,13 @@ const DataRow = ({
   const noBalance = !balance || balance?.equalTo(0)
   const noAllowance = !currentAllowance || currentAllowance.equalTo(0)
 
+  // This is so we only create fiat value request if there is a blance
+  const fiatValue = useMemo(() => {
+    if (!balance && account) return <Loader />
+    else if (noBalance) return <BalanceValue hasBalance={false}>0</BalanceValue>
+    else return <FiatBalanceCell balance={balance} />
+  }, [account, balance, noBalance])
```

Comment:
Nitpicks:

1. Add bracers to if/else
2. Reading this the `noBalance` variable is still confusing, given that we have `!balance` which to me it reads almost the same. What about `hasZeroBalance` instead?
3. Similar to point above and to be consistent, rename `noAllowance` to `hasNoAllowance`

Reply:
Good suggestions